// @medusajs/dashboard/dist/chunk-GLBHPDR4.mjs
import {
  productTypesQueryKeys,
  useInfiniteProductTypes
} from "./chunk-UJ2TMPV4.mjs";
import {
  collectionsQueryKeys,
  useInfiniteCollections
} from "./chunk-UWY5ZV66.mjs";
import {
  ordersQueryKeys
} from "./chunk-4JQR6QNW.mjs";
import {
  categoriesQueryKeys,
  useInfiniteCategories
} from "./chunk-OSHH5GAS.mjs";
import {
  productsQueryKeys,
  useInfiniteProducts
} from "./chunk-Z6BFNHEO.mjs";
import {
  useInfiniteList
} from "./chunk-HGRIOEAR.mjs";
import {
  queryClient
} from "./chunk-FXYH54JP.mjs";
import {
  queryKeysFactory
} from "./chunk-774WSTCC.mjs";
import {
  sdk
} from "./chunk-NFEK63OE.mjs";

// src/hooks/api/plugins.tsx
import { useQuery } from "@tanstack/react-query";
var PLUGINS_QUERY_KEY = "plugins";
var pluginsQueryKeys = queryKeysFactory(PLUGINS_QUERY_KEY);
var usePlugins = (options) => {
  const { data, ...rest } = useQuery({
    queryFn: () => sdk.admin.plugin.list(),
    queryKey: pluginsQueryKeys.list(),
    ...options
  });
  return { ...data, ...rest };
};

// src/hooks/api/product-variants.tsx
import {
  useQuery as useQuery2
} from "@tanstack/react-query";
var PRODUCT_VARIANT_QUERY_KEY = "product_variant";
var productVariantQueryKeys = queryKeysFactory(
  PRODUCT_VARIANT_QUERY_KEY
);
var useVariants = (query, options) => {
  const { data, ...rest } = useQuery2({
    queryFn: () => sdk.admin.productVariant.list(query),
    queryKey: productVariantQueryKeys.list(query),
    ...options
  });
  return { ...data, ...rest };
};
var useInfiniteVariants = (query, options) => {
  return useInfiniteList({
    queryKey: (params) => productVariantQueryKeys.list(params),
    queryFn: async (params) => {
      return await sdk.admin.productVariant.list(params);
    },
    query,
    options
  });
};

// src/hooks/api/refund-reasons.tsx
import {
  useMutation,
  useQuery as useQuery3
} from "@tanstack/react-query";
var REFUND_REASONS_QUERY_KEY = "refund_reasons";
var refundReasonsQueryKeys = queryKeysFactory(REFUND_REASONS_QUERY_KEY);
var useRefundReasons = (query, options) => {
  const { data, ...rest } = useQuery3({
    queryFn: () => sdk.admin.refundReason.list(query),
    queryKey: refundReasonsQueryKeys.list(query),
    ...options
  });
  return { ...data, ...rest };
};
var useRefundReason = (id, query, options) => {
  const { data, ...rest } = useQuery3({
    queryFn: () => sdk.admin.refundReason.retrieve(id, query),
    queryKey: refundReasonsQueryKeys.detail(id),
    ...options
  });
  return { ...data, ...rest };
};
var useCreateRefundReason = (query, options) => {
  return useMutation({
    mutationFn: async (data) => sdk.admin.refundReason.create(data, query),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: refundReasonsQueryKeys.lists()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useUpdateRefundReason = (id, options) => {
  return useMutation({
    mutationFn: async (data) => sdk.admin.refundReason.update(id, data),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: refundReasonsQueryKeys.lists()
      });
      queryClient.invalidateQueries({
        queryKey: refundReasonsQueryKeys.detail(data.refund_reason.id)
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useDeleteRefundReasonLazy = (options) => {
  return useMutation({
    mutationFn: (id) => sdk.admin.refundReason.delete(id),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: refundReasonsQueryKeys.lists()
      });
      queryClient.invalidateQueries({
        queryKey: refundReasonsQueryKeys.details()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};

// src/hooks/api/shipping-option-types.tsx
import {
  useMutation as useMutation2,
  useQuery as useQuery4
} from "@tanstack/react-query";
var SHIPPING_OPTION_TYPES_QUERY_KEY = "shipping_option_types";
var shippingOptionTypesQueryKeys = queryKeysFactory(
  SHIPPING_OPTION_TYPES_QUERY_KEY
);
var useShippingOptionType = (id, query, options) => {
  const { data, ...rest } = useQuery4({
    queryFn: () => sdk.admin.shippingOptionType.retrieve(id, query),
    queryKey: shippingOptionTypesQueryKeys.detail(id),
    ...options
  });
  return { ...data, ...rest };
};
var useShippingOptionTypes = (query, options) => {
  const { data, ...rest } = useQuery4({
    queryFn: () => sdk.admin.shippingOptionType.list(query),
    queryKey: shippingOptionTypesQueryKeys.list(query),
    ...options
  });
  return { ...data, ...rest };
};
var useCreateShippingOptionType = (options) => {
  return useMutation2({
    mutationFn: (payload) => sdk.admin.shippingOptionType.create(payload),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: shippingOptionTypesQueryKeys.lists()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useUpdateShippingOptionType = (id, options) => {
  return useMutation2({
    mutationFn: (payload) => sdk.admin.shippingOptionType.update(id, payload),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: shippingOptionTypesQueryKeys.detail(id)
      });
      queryClient.invalidateQueries({
        queryKey: shippingOptionTypesQueryKeys.lists()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useDeleteShippingOptionType = (id, options) => {
  return useMutation2({
    mutationFn: () => sdk.admin.shippingOptionType.delete(id),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: shippingOptionTypesQueryKeys.detail(id)
      });
      queryClient.invalidateQueries({
        queryKey: shippingOptionTypesQueryKeys.lists()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};

// src/hooks/api/tags.tsx
import {
  useMutation as useMutation3,
  useQuery as useQuery5
} from "@tanstack/react-query";
var TAGS_QUERY_KEY = "tags";
var productTagsQueryKeys = queryKeysFactory(TAGS_QUERY_KEY);
var useProductTag = (id, query, options) => {
  const { data, ...rest } = useQuery5({
    queryKey: productTagsQueryKeys.detail(id, query),
    queryFn: async () => sdk.admin.productTag.retrieve(id),
    ...options
  });
  return { ...data, ...rest };
};
var useProductTags = (query, options) => {
  const { data, ...rest } = useQuery5({
    queryKey: productTagsQueryKeys.list(query),
    queryFn: async () => sdk.admin.productTag.list(query),
    ...options
  });
  return { ...data, ...rest };
};
var useInfiniteProductTags = (query, options) => {
  return useInfiniteList({
    queryKey: (params) => productTagsQueryKeys.list(params),
    queryFn: (params) => sdk.admin.productTag.list(params),
    query,
    options
  });
};
var useCreateProductTag = (query, options) => {
  return useMutation3({
    mutationFn: async (data) => sdk.admin.productTag.create(data, query),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: productTagsQueryKeys.lists()
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useUpdateProductTag = (id, query, options) => {
  return useMutation3({
    mutationFn: async (data) => sdk.admin.productTag.update(id, data, query),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: productTagsQueryKeys.lists()
      });
      queryClient.invalidateQueries({
        queryKey: productTagsQueryKeys.detail(data.product_tag.id, query)
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useDeleteProductTag = (id, options) => {
  return useMutation3({
    mutationFn: async () => sdk.admin.productTag.delete(id),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: productTagsQueryKeys.lists()
      });
      queryClient.invalidateQueries({
        queryKey: productTagsQueryKeys.detail(id)
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};

// src/hooks/api/translations.tsx
import {
  useMutation as useMutation4,
  useQuery as useQuery6
} from "@tanstack/react-query";
var TRANSLATIONS_QUERY_KEY = "translations";
var translationsQueryKeys = queryKeysFactory(TRANSLATIONS_QUERY_KEY);
var TRANSLATION_SETTINGS_QUERY_KEY = "translation_settings";
var translationSettingsQueryKeys = queryKeysFactory(
  TRANSLATION_SETTINGS_QUERY_KEY
);
var TRANSLATION_STATISTICS_QUERY_KEY = "translation_statistics";
var translationStatisticsQueryKeys = queryKeysFactory(
  TRANSLATION_STATISTICS_QUERY_KEY
);
var useReferenceTranslations = (reference, translatableFields, referenceId, options) => {
  const referenceHookMap = /* @__PURE__ */ new Map([
    [
      "product",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteProducts(
          { fields, id: referenceId ?? [] },
          options
        );
        const products = data2?.pages.flatMap((page) => page.products) ?? [];
        return {
          ...rest2,
          data: {
            translations: products?.flatMap((product) => product.translations ?? []) ?? [],
            references: products ?? [],
            count: data2?.pages[0]?.count ?? 0
          }
        };
      }
    ],
    [
      "product_variant",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteVariants(
          { id: referenceId ?? [], fields },
          options
        );
        const variants = data2?.pages.flatMap((page) => page.variants) ?? [];
        return {
          ...rest2,
          data: {
            translations: variants?.flatMap((variant) => variant.translations ?? []) ?? [],
            references: variants ?? [],
            translatableFields,
            count: data2?.pages[0]?.count ?? 0
          }
        };
      }
    ],
    [
      "product_category",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteCategories(
          { id: referenceId ?? [], fields },
          options
        );
        const categories = data2?.pages.flatMap((page) => page.product_categories) ?? [];
        return {
          ...rest2,
          data: {
            translations: categories?.flatMap((category) => category.translations ?? []) ?? [],
            references: categories ?? [],
            translatableFields,
            count: data2?.pages[0]?.count ?? 0
          }
        };
      }
    ],
    [
      "product_collection",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteCollections(
          { id: referenceId ?? [], fields },
          options
        );
        const collections = data2?.pages.flatMap((page) => page.collections) ?? [];
        return {
          ...rest2,
          data: {
            translations: collections?.flatMap(
              (collection) => collection.translations ?? []
            ) ?? [],
            references: collections ?? [],
            translatableFields,
            count: data2?.pages[0]?.count ?? 0
          }
        };
      }
    ],
    [
      "product_type",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteProductTypes(
          { id: referenceId ?? [], fields },
          options
        );
        const product_types = data2?.pages.flatMap((page) => page.product_types) ?? [];
        return {
          ...rest2,
          data: {
            translations: product_types?.flatMap((type) => type.translations ?? []) ?? [],
            references: product_types ?? [],
            count: data2?.pages[0]?.count ?? 0,
            translatableFields
          }
        };
      }
    ],
    [
      "product_tag",
      () => {
        const fields = translatableFields.concat(["translations.*"]).join(",");
        const { data: data2, ...rest2 } = useInfiniteProductTags(
          { id: referenceId ?? [], fields },
          options
        );
        const product_tags = data2?.pages.flatMap((page) => page.product_tags) ?? [];
        return {
          ...rest2,
          data: {
            translations: product_tags?.flatMap((tag) => tag.translations ?? []) ?? [],
            references: product_tags ?? [],
            translatableFields,
            count: data2?.pages[0]?.count ?? 0
          }
        };
      }
    ]
    // TODO: product option and option values
  ]);
  const referenceHook = referenceHookMap.get(reference);
  if (!referenceHook) {
    throw new Error(`No hook found for reference type: ${reference}`);
  }
  const { data, ...rest } = referenceHook();
  return { ...data, ...rest };
};
var referenceInvalidationKeysMap = /* @__PURE__ */ new Map([
  ["product", productsQueryKeys.lists()],
  ["product_variant", productVariantQueryKeys.lists()],
  ["product_category", categoriesQueryKeys.lists()],
  ["product_collection", collectionsQueryKeys.lists()],
  ["product_type", productTypesQueryKeys.lists()],
  ["product_tag", productTagsQueryKeys.lists()]
]);
var useBatchTranslations = (reference, options) => {
  const mutation = useMutation4({
    mutationFn: (payload) => sdk.admin.translation.batch(payload),
    ...options
  });
  const invalidateQueries = async () => {
    await Promise.all([
      queryClient.invalidateQueries({
        queryKey: referenceInvalidationKeysMap.get(reference)
      }),
      queryClient.invalidateQueries({
        queryKey: translationStatisticsQueryKeys.lists()
      })
    ]);
  };
  return {
    ...mutation,
    invalidateQueries
  };
};
var useTranslationSettings = (query, options) => {
  const { data, ...rest } = useQuery6({
    queryKey: translationSettingsQueryKeys.list(query),
    queryFn: () => sdk.admin.translation.settings(query),
    ...options
  });
  return { ...data, ...rest };
};
var useTranslationStatistics = (query, options) => {
  const { data, ...rest } = useQuery6({
    queryKey: translationStatisticsQueryKeys.list(query),
    queryFn: () => sdk.admin.translation.statistics(query),
    ...options
  });
  return { ...data, ...rest };
};

// src/hooks/api/fulfillment.tsx
import { useMutation as useMutation5 } from "@tanstack/react-query";
var FULFILLMENTS_QUERY_KEY = "fulfillments";
var fulfillmentsQueryKeys = queryKeysFactory(FULFILLMENTS_QUERY_KEY);

// src/hooks/api/locales.tsx
import { useQuery as useQuery7 } from "@tanstack/react-query";
var LOCALES_QUERY_KEY = "locales";
var localesQueryKeys = queryKeysFactory(LOCALES_QUERY_KEY);
var useLocales = (query, options) => {
  const { data, ...rest } = useQuery7({
    queryFn: () => sdk.admin.locale.list(query),
    queryKey: localesQueryKeys.list(query),
    ...options
  });
  return { ...data, ...rest };
};

// src/hooks/api/notification.tsx
import { useQuery as useQuery8 } from "@tanstack/react-query";
var NOTIFICATION_QUERY_KEY = "notification";
var notificationQueryKeys = queryKeysFactory(NOTIFICATION_QUERY_KEY);
var useNotifications = (query, options) => {
  const { data, ...rest } = useQuery8({
    queryFn: () => sdk.admin.notification.list(query),
    queryKey: notificationQueryKeys.list(query),
    ...options
  });
  return { ...data, ...rest };
};

// src/hooks/api/payment-collections.tsx
import { useMutation as useMutation6 } from "@tanstack/react-query";
var PAYMENT_COLLECTION_QUERY_KEY = "payment-collection";
var paymentCollectionQueryKeys = queryKeysFactory(
  PAYMENT_COLLECTION_QUERY_KEY
);
var useMarkPaymentCollectionAsPaid = (orderId, paymentCollectionId, options) => {
  return useMutation6({
    mutationFn: (payload) => sdk.admin.paymentCollection.markAsPaid(paymentCollectionId, payload),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.details()
      });
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.preview(orderId)
      });
      queryClient.invalidateQueries({
        queryKey: paymentCollectionQueryKeys.all
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};

// src/hooks/api/payments.tsx
import {
  useMutation as useMutation7,
  useQuery as useQuery9
} from "@tanstack/react-query";
var PAYMENT_QUERY_KEY = "payment";
var paymentQueryKeys = queryKeysFactory(PAYMENT_QUERY_KEY);
var PAYMENT_PROVIDERS_QUERY_KEY = "payment_providers";
var paymentProvidersQueryKeys = queryKeysFactory(
  PAYMENT_PROVIDERS_QUERY_KEY
);
var useCapturePayment = (orderId, paymentId, options) => {
  return useMutation7({
    mutationFn: (payload) => sdk.admin.payment.capture(paymentId, payload),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.details()
      });
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.preview(orderId)
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};
var useRefundPayment = (orderId, paymentId, options) => {
  return useMutation7({
    mutationFn: (payload) => sdk.admin.payment.refund(paymentId, payload),
    onSuccess: (data, variables, context) => {
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.details()
      });
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.preview(orderId)
      });
      options?.onSuccess?.(data, variables, context);
    },
    ...options
  });
};

export {
  useLocales,
  notificationQueryKeys,
  useNotifications,
  useMarkPaymentCollectionAsPaid,
  useCapturePayment,
  useRefundPayment,
  usePlugins,
  productVariantQueryKeys,
  useVariants,
  refundReasonsQueryKeys,
  useRefundReasons,
  useRefundReason,
  useCreateRefundReason,
  useUpdateRefundReason,
  useDeleteRefundReasonLazy,
  shippingOptionTypesQueryKeys,
  useShippingOptionType,
  useShippingOptionTypes,
  useCreateShippingOptionType,
  useUpdateShippingOptionType,
  useDeleteShippingOptionType,
  productTagsQueryKeys,
  useProductTag,
  useProductTags,
  useCreateProductTag,
  useUpdateProductTag,
  useDeleteProductTag,
  useReferenceTranslations,
  useBatchTranslations,
  useTranslationSettings,
  useTranslationStatistics
};
