// @medusajs/medusa/dist/api/auth/utils/generate-jwt-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateJwtTokenForAuthIdentity = generateJwtTokenForAuthIdentity;
const utils_1 = require("@medusajs/framework/utils");
function generateJwtTokenForAuthIdentity({ authIdentity, actorType, authProvider, }, { secret, expiresIn, options, }) {
    const expiresIn_ = expiresIn ?? options?.expiresIn;
    const entityIdKey = `${actorType}_id`;
    const entityId = authIdentity?.app_metadata?.[entityIdKey];
    const providerIdentity = !authProvider
        ? undefined
        : authIdentity.provider_identities?.filter((identity) => identity.provider === authProvider)[0];
    return (0, utils_1.generateJwtToken)({
        actor_id: entityId ?? "",
        actor_type: actorType,
        auth_identity_id: authIdentity?.id ?? "",
        app_metadata: {
            [entityIdKey]: entityId,
        },
        user_metadata: providerIdentity?.user_metadata ?? {},
    }, {
        secret,
        expiresIn: expiresIn_,
        jwtOptions: options,
    });
}
